pkgs/desktops/gnome-3/misc/nautilus-python/default.nix

```diff
@@ -0,0 +1,47 @@
+{ stdenv, fetchFromGitLab
+, automake, autoconf, libtool, pkgconfig, which, gnome-common, gtk-doc, file
+, python3, python3Packages, nautilus }:
+
+stdenv.mkDerivation rec {
+  name = "nautilus-python-${version}";
+  version = "1.2.2";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "GNOME";
+    repo = "nautilus-python";
+    rev = "44224fac32d2d5284956b55ecd884fdfcb1bec4e";
+    sha256 = "1473kr1zm2cm8pxg1nhp6j27yh7i1q5qgz6g6lb57kvj7sf5vl8f";
+  };
+
+  nativeBuildInputs = [
+    automake autoconf libtool pkgconfig which gnome-common gtk-doc
+  ];
+
+  buildInputs = [
+    python3 python3Packages.pygobject3 python3 nautilus
+  ];
+
+  # checking for python version... 3.6
+  # checking for python platform... linux
+  # checking for python script directory... ${prefix}/lib/python3.6/site-packages
+  # checking for python extension module directory... ${exec_prefix}/lib/python3.6/site-packages
+  # checking for headers required to compile python extensions... found
+  # checking for libraries required to embed python... no
+  # configure: error: could not find Python lib
+  configurePhase = ''
+    NOCONFIGURE=1 ./autogen.sh
```

Comment:
This should generally go to `preConfigure`, though it is not needed with the dist tarball.